src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -396,31 +396,23 @@ private async Task<SerializableDiagnosticAnalysisResults> AnalyzeAsync(
                 if (documentAnalysisScope == null)
                     unitCount += _project.DocumentIds.Count;
 
-                var projectPerformanceInfo = projectAnalysisResult?.AnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(_analyzerInfoCache) ?? [];
-                var hostPerformanceInfo = hostAnalysisResult?.AnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(_analyzerInfoCache) ?? [];
-                _performanceTracker.AddSnapshot(projectPerformanceInfo.Concat(hostPerformanceInfo), unitCount, forSpanAnalysis: _span.HasValue);
+                var performanceInfo = analysisResult?.MergedAnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(_analyzerInfoCache) ?? [];
+                _performanceTracker.AddSnapshot(performanceInfo, unitCount, forSpanAnalysis: _span.HasValue);
             }
         }
 
         var builderMap = ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResultBuilder>.Empty;
-        if (projectAnalysisResult is not null)
+        if (analysisResult is not null)
         {
-            builderMap = builderMap.AddRange(await projectAnalysisResult.ToResultBuilderMapAsync(
+            builderMap = builderMap.AddRange(await analysisResult.ToResultBuilderMapAsync(
                 additionalPragmaSuppressionDiagnostics, documentAnalysisScope,
-                _project, VersionStamp.Default, compilationWithAnalyzers.ProjectCompilation!,
-                projectAnalyzers, skippedAnalyzersInfo, reportSuppressedDiagnostics, cancellationToken).ConfigureAwait(false));
-        }
-
-        if (hostAnalysisResult is not null)
-        {
-            builderMap = builderMap.AddRange(await hostAnalysisResult.ToResultBuilderMapAsync(
-                additionalPragmaSuppressionDiagnostics, documentAnalysisScope,
-                _project, VersionStamp.Default, compilationWithAnalyzers.HostCompilation!,
-                hostAnalyzers, skippedAnalyzersInfo, reportSuppressedDiagnostics, cancellationToken).ConfigureAwait(false));
+                _project, VersionStamp.Default, compilationWithAnalyzers.ProjectCompilation, compilationWithAnalyzers.HostCompilation,
+                projectAnalyzers, hostAnalyzers, skippedAnalyzersInfo, reportSuppressedDiagnostics, cancellationToken).ConfigureAwait(false));
         }
 
+        // TODO Does this need adjustment?
```

Comment:
📝 Need to trace this data back and see if the `telemetry` list is ever converted back to a dictionary

Reply:
This is converted back to a dictionary, so I need to modify the code to avoid having the key present twice.

Reply:
➡️ Problems are now avoided by using the merged telemetry dictionary